### DISCUSSION TITLE
Create3 used to work w/ Bluetooth, now doesn't show up at all

### DISCUSSION THREAD

**Discussion by monkey-w1n5t0n** [iRobotEducation/create3_docs#574] [Questions]
### How are you connecting to your Create 3?

Bluetooth (Python SDK)

### Computer(s) Model(s) and Operating System(s)

Raspberry Pi (Ubuntu 22.04 LTS), Arch Linux

### Which version of ROS 2 is installed on your computer?

None

### Which firmware version is installed on your robot?

H.2.6

### Which RMW is your robot running?

FastRTPS

### Is the robot connected to a network? If so, what is the network type?

Yes, offline LAN (also tried with mobile hotspot)

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

Bluetooth (default)

### Describe your question.

This particular Create3 was successfully used over Bluetooth (Python SDK) recently. Now it won't show up on any Bluetooth device (it used to) and the Python script won't connect.

Re-flashed firmware, changed the Bluetooth name, did full reboots, switched back and forth between USB and Bluetooth, blue light has been on the entire time, removed and re-inserted the Bluetooth board - no luck.

Thoughts as to what may be the problem?

Logs include the following lines that seem relevant:
```
Mar 26 13:55:03 PIBOT3 daemon.info bluetoothd[1213]: Bluetooth daemon 5.63
Mar 26 13:55:03 PIBOT3 kern.info kernel: [   42.257420] asoc-simple-card sound@1: HiFi <-> 10000a00.i2s mapping ok
Mar 26 13:55:03 PIBOT3 kern.notice kernel: [   42.512083] random: bluetoothd: uninitialized urandom read (4 bytes read)
Mar 26 13:55:03 PIBOT3 daemon.info bluetoothd[1213]: Bluetooth management interface 1.14 initialized
Mar 26 13:55:03 PIBOT3 daemon.err bluetoothd[1213]: ../bluez-5.63/src/adapter.c:reset_adv_monitors_complete() Failed to reset Adv Monitors: Unknown Command (0x01)
Mar 26 13:55:03 PIBOT3 daemon.info bluetoothd[1213]: Battery Provider Manager created
Mar 26 13:55:03 PIBOT3 daemon.err bluetoothd[1213]: Failed to Read Adv Monitor Features with status 0x01
```

I'm attaching the logs from the robot.
[messages.txt](https://github.com/iRobotEducation/create3_docs/files/14761361/messages.txt)
[messages(1).txt](https://github.com/iRobotEducation/create3_docs/files/14761367/messages.1.txt)

**Comment by shamlian** [comment]
Thanks for sending logs. From all appearances, the robot sees the Bluetooth adapter and is starting up correctly. Out of curiosity, what operating system is the computer you are using to talk to the robot? If Linux with BlueZ, I wonder if you are encountering this BlueZ bug (what version of bluetoothd are you running)? https://github.com/iRobotEducation/irobot-edu-python-sdk/issues/3

**Reply by monkey-w1n5t0n** [reply]
Tried both Manjaro and Ubuntu 22.04 LTS, both of which were being used successfully used with the very same robot not long ago.

The Manjaro system is on bluez 5.72-2.

The robot's Bluetooth name definitely used to appear on both the Manjaro and Ubuntu systems (same installation) as soon as the robot booted and without any further prompting, but now does not appear on any bluetooth-enabled device, including phones etc.

**Reply by shamlian** [reply]
I believe you, just trying to figure out what's going on from across an Ethernet port.

The robot uses BLE, so you would usually only see if it you were trying to access it with something specifically looking for the iRobot BLE protocol, like the iRobot Coding app -- it would not appear as a Bluetooth device in your settings on a phone. (I guess you could also see it in a BLE sniffer like LightBlue or nRF Connect.) What does `bluetoothctl devices` yield on one of your Linux machines?

**Reply by shamlian** [reply]
Oh -- another thing that seems odd that I just noticed -- you said you switched between USB and BLE and the blue light stayed on the entire time. That's not expected; the blue light should turn off when you flip the switch from BLE to USB. I wonder if there is some kind of weird hardware fault? Can you verify that flipping the switch to USB does not extinguish the LED? If it does not, we might need to ship you a new adapter board (for that, you should email education@irobot.com, referencing this discussion).

**Comment by monkey-w1n5t0n** [comment]
> The robot uses BLE, so you would usually only see if it you were trying to access it with something specifically looking for the iRobot BLE protocol, like the iRobot Coding app -- it would not appear as a Bluetooth device in your settings on a phone. 

I thought so, but I used to see it normally in blueman before, which is why I found it significant that it's not showing up now. After a brief period of inactivity, I managed to connect briefly with my laptop and send it some navigation commands. A couple of days later I tried to connect it to a Raspberry Pi 4, it briefly (~2s) showed up on the Bluetooth devices list and then disappeared without any further action on my part (e.g. trying to connect), and hasn't showed back since.

> What does `bluetoothctl devices` yield on one of your Linux machines?

It shows a couple of unrelated devices (the same ones I can see in blueman), which do not include the Create3.

> you said you switched between USB and BLE and the blue light stayed on the entire time

Sorry, my bad for phrasing it this way: the blue light is definitely extinguished when it's switched to USB, I just meant that at no point did the light fail to light up when it should (as I've read a couple reports of people having that issue).

**Reply by shamlian** [reply]
This is quite strange; if after a cold boot with the BLE adapter on, it does not advertise, I think the most efficient thing to do would be to send you a new adapter board to rule out a hardware problem. A brief test with my Linux machine with BlueZ 5.64 and robot running H.2.6 seem to be working as normal. I will find some time later this week to see if I can torture the robot into a bad state to reproduce your problem, in the case there is a bug to solve.

**Comment by monkey-w1n5t0n** [comment]
Thanks, I've followed up via email. Shall we leave this discussion open until we have verified that it was a hardware issue?

**Reply by shamlian** [reply]
Yes, please do.

**Comment by sleeplessTLV** [comment]
Hi, my BT connection seems to have died (no cold restart, connector boards check helps). WiFi connection still fine and robot does not even report an BT errors in the log.
Can I still use it? for control via the Python SDK?
